examples/CMakeLists.txt

```diff
@@ -21,6 +21,23 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 # IN THE SOFTWARE.
 ###############################################################################
+cmake_minimum_required(VERSION 3.21 FATAL_ERROR)
+
+###############################################################################
+# DEFAULT BUILD TYPE
+###############################################################################
+set(CMAKE_BUILD_TYPE "Release" CACHE STRING
+      "build type: Release, Debug, RelWithDebInfo, MinSizeRel")
+
+add_compile_options($<$<CONFIG:Debug>:-ggdb>)
+add_link_options($<$<CONFIG:Debug>:-ggdb>)
+
+set(CMAKE_HIP_ARCHITECTURES OFF)
+
+###############################################################################
+# PROJECT
+###############################################################################
+project(rocshmem_examples VERSION 1.0.0 LANGUAGES HIP CXX)
```

Comment:
trying this approach with using LANGUAGE HIP, one open question is how much control we have with choosing the hip/rocm version this way.